examples/src/bin/tlsserver-mio.rs

```diff
@@ -417,79 +416,65 @@ impl OpenConnection {
-    --require-auth      Send a fatal alert if the client does not complete client
-                        authentication.
-    --no-resumption     Disable stateful session resumption.
-    --tickets           Support tickets (stateless resumption).
-    --max-early-data BYTES   Support receiving BYTES many bytes with 0RTT
-    --protover VERSION  Disable default TLS version list, and use
-                        VERSION instead.  May be used multiple times.
-    --suite SUITE       Disable default cipher suite list, and use
-                        SUITE instead.  May be used multiple times.
-    --proto PROTOCOL    Negotiate PROTOCOL using ALPN.
-                        May be used multiple times.
-    --verbose           Emit log output.
-    --version, -v       Show tool version.
-    --help, -h          Show this screen.
-";
-
-#[derive(Debug, Deserialize)]
+/// Runs a TLS server on :PORT.  The default PORT is 443.
+///
+/// `echo' mode means the server echoes received data on each connection.
+///
+/// `http' mode means the server blindly sends a HTTP response on each connection.
+///
+/// `forward' means the server forwards plaintext to a connection made to `localhost:fport`.
+///
+/// `--certs' names the full certificate chain, `--key' provides the private key.
+#[derive(Debug, Parser)]
 struct Args {
-    cmd_echo: bool,
-    cmd_http: bool,
-    flag_port: Option<u16>,
-    flag_verbose: bool,
-    flag_protover: Vec<String>,
-    flag_suite: Vec<String>,
-    flag_proto: Vec<String>,
-    flag_certs: Option<String>,
-    flag_crl: Vec<String>,
-    flag_key: Option<String>,
-    flag_ocsp: Option<String>,
-    flag_auth: Option<String>,
-    flag_require_auth: bool,
-    flag_no_resumption: bool,
-    flag_tickets: bool,
-    flag_max_early_data: u32,
-    arg_fport: Option<u16>,
+    #[command(subcommand)]
+    cmd: ServerMode,
+    /// Listen on port.
+    #[clap(short, long, default_value = "443")]
+    port: u16,
+    /// Emit log output.
+    #[clap(short, long)]
+    verbose: bool,
+    /// Disable default TLS version list, and use the given versions instead.
+    #[clap(long)]
+    protover: Vec<String>,
+    /// Disable default cipher suite list, and use the given suites instead.
+    #[clap(long)]
+    suite: Vec<String>,
+    /// Negotiate the given protocols using ALPN.
+    #[clap(long)]
+    proto: Vec<String>,
+    /// Read server certificates from the given file. This should contain PEM-format certificates
+    /// in the right order (the first certificate should certify the end entity, matching the
+    /// private key, the last should be a root CA).
+    #[clap(long)]
+    certs: Option<String>,
+    /// Perform client certificate revocation checking using the DER-encoded CRL from the given
+    /// files.
+    #[clap(long)]
+    crl: Vec<String>,
+    /// Read private key from the given file. This should be a private key in PEM format.
+    #[clap(long)]
+    key: Option<String>,
```

Comment:
```suggestion
    key: String,
```
(i think this should be mandatory, to get clap to generate the error message?)

Reply:
if we do that, we can get rid of `args.key.as_ref().expect("--key option missing")` below